### ISSUE TITLE
Mouse-Related events aren't overrideable like KeyEvents are

### ISSUE THREAD

**Issue by JordanMartinez** [TomasMikula/RichTextFX#217]
In `StyledTextAreaBehavior`, `MouseEvent`s are handled using `EventStream`s which don't take into consideration if the `MouseEvent` is already consumed (unlike `EventHandlerTemplate` and `KeyEvent`s):

````java
EventHandlerHelper.installAfter(area.onKeyPressedProperty(), keyPressedHandler);
EventHandlerHelper.installAfter(area.onKeyTypedProperty(), keyTypedHandler);

subscription = Subscription.multi(
        eventsOf(area, MouseEvent.MOUSE_PRESSED).subscribe(this::mousePressed),
        eventsOf(area, MouseEvent.MOUSE_DRAGGED).subscribe(this::mouseDragged),
        eventsOf(area, MouseEvent.DRAG_DETECTED).subscribe(this::dragDetected),
        eventsOf(area, MouseEvent.MOUSE_RELEASED).subscribe(this::mouseReleased),
        () -> {
            EventHandlerHelper.remove(area.onKeyPressedProperty(), keyPressedHandler);
            EventHandlerHelper.remove(area.onKeyTypedProperty(), keyTypedHandler);
        });
````

The following summarized code doesn't work:
````groovy
EventHandler<MouseEvent> mouseRelease = EventHandlerTemplate
            .on(MouseEvent.MOUSE_RELEASED).act {area, event -> event.consume() }
            .create()
            .bind(this)
EventHandlerHelper.install(onMouseReleasedProperty(), mouseRelease)
````
If I want to customize the MouseEvent handling, it can only be done by adding an `EventFilter`

**Comment by TomasMikula** [comment]
Good point. We should investigate how to port the mouse event handling to the same approach that is used for key handling.

**Comment by TomasMikula** [comment]
Oh, I see you already did that :)

**Comment by JordanMartinez** [comment]
:smile: Yup!

**Comment by JordanMartinez** [comment]
So...... this issue and others I've opened (#218, #223, #152) have been resolved in the current `master` branch but haven't yet been included in an official release. Am I supposed to close them now, or wait for you to do so when the next version that includes the code that resolves these issues is released?

**Comment by TomasMikula** [comment]
Thanks for checking up on them. I think we can close them now :)

**Comment by JordanMartinez** [comment]
Resolved via ca3b771cb00cc7b051dcd1080ef5e8ba9a2df140
